ibc-clients/cw-context/src/context/mod.rs

```diff
@@ -18,10 +18,15 @@ use prost::Message;
 
 use crate::api::ClientType;
 use crate::types::{ContractError, GenesisMetadata, HeightTravel, MigrationPrefix};
-use crate::utils::{parse_height, AnyCodec};
+use crate::utils::AnyCodec;
+use cosmwasm_std::Empty;
+use cw_storage_plus::{Bound, Map};
 
 type Checksum = Vec<u8>;
 
+pub const CONSENSUS_STATE_HEIGHT_MAP: Map<'_, (u64, u64), Empty> =
+    Map::new(ITERATE_CONSENSUS_STATE_PREFIX);
```

Comment:
I started using this for a better interface. I am curious - if anything breaks if I do this.

Reply:
Note that we only want to maintain a sorted set. So the key is set to [Empty](https://book.cosmwasm.com/cross-contract/map-storage.html#maps-as-sets).

Reply:
Cool. We should be good as long as the key produced by the `Map` matches the `iteration_key()` (?)
Specifically, we only need to use the `iteration_key` [here](https://github.com/cosmos/ibc-rs/blob/4a4679c3149e69254b77d3f4b19a3d27c4b2cac5/ibc-clients/cw-context/src/context/mod.rs#L237). (Connected to your question [below](https://github.com/cosmos/ibc-rs/pull/1176#discussion_r1573240899))

Reply:
I realized in the old implementation, we should have used this [`range_with_prefix`](https://docs.rs/cw-storage-plus/1.0.1/src/cw_storage_plus/prefix.rs.html#193) from cw-storage-plus. Note the trick for `None` and exclusive or inclusive cases in [`calc_start_bound`](https://docs.rs/cw-storage-plus/1.0.1/src/cw_storage_plus/prefix.rs.html#214) and [`calc_end_bound`](https://docs.rs/cw-storage-plus/1.0.1/src/cw_storage_plus/prefix.rs.html#223).

Since this brings `cw-storage-plus` dependency anyway, it's best to use their storage abstractions, built on top of these low-level functions.